src/arrayMethodJoin.js

```diff
@@ -5,7 +5,28 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let sep = separator;
+    let result = '';
+
+    if (separator === undefined) {
+      sep = ',';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        if (i !== this.length - 1) {
```

Comment:
Avoid using nested `if`. Try to refactor it to improve the readability. Use a ternary operator, maybe. Ask questions in the chat, if you need to